### PR TITLE
build: Yarn 3 lifecycle scripts

### PR DESCRIPTION
**Pull request by marafat**
## 💡 Context

Yarn 3 doesn't run `prapare` npm lifecycle event, [docs](https://yarnpkg.com/advanced/lifecycle-scripts). So husky git hooks doesn't get installed when someone checks out the repo to contribute.

## 🎉 Changes

Following the steps from [husky docs for Yarn 2](https://typicode.github.io/husky/#/?id=install-1) (also applies to Yarn 3):

- introduce `postinstall` with yarn 3 cleanup and husky install
- remove `prepare`
- diable `postinstall` in `prepack` and enable it in `postpack` (as noted in the husky docs)

## ✔️ Checklist

- [x] Code is safe to be deployed to production
- [ ] Tests added for all changes

### CODE REVIEW

**Review by vanstinator — DISMISSED** [review]
Nice find

File: .gitignore [hunk 1]
```diff
@@ -14,4 +14,7 @@ yarn-error.log
 !.yarn/plugins
 !.yarn/releases
 !.yarn/sdks
-!.yarn/versions
\ No newline at end of file
+!.yarn/versions
+
+# Webstorm
```
Comment (ejmartin504) [inline .gitignore:19]: 😂

File: package.json [hunk 2]
```diff
@@ -42,8 +42,9 @@
     "docs": "node docs/build.js && docsify serve ./docs",
     "benchmark": "yarn build && node --experimental-specifier-resolution=node ./benchmark/run.js",
     "lint": "eslint --ext .js,.mjs,.ts .",
-    "prepack": "yarn build",
-    "prepare": "husky install",
+    "postinstall": "yarn clean-modules -f .yarnclean -y && husky install",
+    "postpack": "pinst --enable",
```
Comment (avaly) [inline package.json:46]: I think we need to install this package in our dev dependencies, right?

**Review by avaly — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (marafat) [reply package.json:46]: Nice catch!! The docs didn't mention it as an additional npm repo, and I just assumed it was some npm _magic_ 😅

**Review by avaly — APPROVED** [review]

**Pull request merged**